terraform/variables.tf

```diff
@@ -17,37 +17,80 @@ variable "project_id" {
   type        = string
 }
 
-variable "name" {
-  description = "The name of this component."
+variable "prefix_name" {
+  description = "The prefix applied to all components."
   type        = string
+  default     = "github-metrics"
   validation {
-    condition     = can(regex("^[A-Za-z][0-9A-Za-z-]+[0-9A-Za-z]$", var.name))
+    condition     = can(regex("^[A-Za-z][0-9A-Za-z-]+[0-9A-Za-z]$", var.prefix_name))
     error_message = "Name can only contain letters, numbers, hyphens(-) and must start with letter."
   }
 }
 
-variable "domain" {
-  description = "Domain name for the Google Cloud Load Balancer."
+variable "component_names" {
```

Comment:
I might drop this, and just hard code throughout, use the prefix_name to allow the user to provide their name, for example how the pubsub.tf components are using it.